bin/improver-standardise

```diff
@@ -70,39 +72,56 @@ def main():
                              "output file, then the source will be checked"
                              "for float64 data.")
 
-    parser.add_argument("--target_grid_filepath", metavar="TARGET_GRID",
-                        help='If specified then regridding of the source '
-                             'against the target grid is enabled.')
+    regrid_group = parser.add_argument_group("Regridding options")
+    regrid_group.add_argument(
+        "--target_grid_filepath", metavar="TARGET_GRID",
+        help=('If specified then regridding of the source '
+              'against the target grid is enabled. If also using '
+              'landmask-aware regridding, then this must be land_binary_mask '
+              'data.'))
+
+    regrid_group.add_argument(
+        "--nearest", action='store_true', default=False,
+        help='If True, regridding will be performed using '
+             'iris.analysis.Nearest() instead of Linear(). '
+             'Use for less continuous fields, '
+             'e.g. precipitation.')
+
+    regrid_group.add_argument(
+        "--extrapolation_mode", default='nanmask',
+        help='Mode to use for extrapolating data into regions '
+             'beyond the limits of the source_data domain. '
+             'Modes are: '
+             'extrapolate - The extrapolation points will '
+             'take their value from the nearest source point. '
+             'nan - The extrapolation points will be be '
+             'set to NaN. '
+             'error - A ValueError exception will be raised, '
+             'notifying an attempt to extrapolate. '
+             'mask  - The extrapolation points will always be '
+             'masked, even if the source data is not a '
+             'MaskedArray. '
+             'nanmask - If the source data is a MaskedArray '
+             'the extrapolation points will be masked. '
+             'Otherwise they will be set to NaN. '
+             'Defaults to nanmask.')
+
+    regrid_group.add_argument(
```

Comment:
I think these two places are the only point you actually say we can do coastline aware regridding. I think it probably needs to be a bit clearer somewhere that there are 3 types of regridding, land-sea aware, nearest and bilinear. The clearest solution might be to change the --nearest flag to be a --regridding_method option, which has to take one of the 3 options. Then you're not just triggering land-sea regridding based on whether an the extra land-sea mask file is passed in.

Reply:
Replaced --nearest with --regrid_mode.